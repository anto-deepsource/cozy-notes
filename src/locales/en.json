{
  "Notes": {
    "Add": {
      "add_note": "Add a note"
    },
    "Delete": {
      "delete_note": "Delete",
      "deleted": "The note has been moved to the trash of your Drive",
      "failed": "The note could not be deleted, please try again"
    },
    "EditorView": {
      "main_placeholder": "Write anythingâ€¦"
    },
    "Editor": {
      "title_placeholder": "Here is your title",
      "exit_confirmation_message": "Automatic saving has not taken into account the latest changes in your doc. Please stay on your current note if you want to keep them.",
      "exit_confirmation_title": "Abandon your changes?",
      "exit_confirmation_leave": "Leave without saving",
      "exit_confirmation_cancel": "Cancel"
    },
    "List": {
      "my_notes": "My Notes",
      "name": "Name",
      "updated_at": "Updated at",
      "location": "Location",
      "sharings": "Sharings",
      "at": "%{date} at %{time}"
    },
    "Empty": {
      "welcome": "Welcome on Cozy Notes",
      "after_created": "Once your Note is created, you can find it in your "
    },
    "SavingIndicator": {
      "saved_just_now": "Saved just now.",
      "saved_secs_ago": "Saved %{time} sec ago.",
      "saved_min_ago": "Saved %{time} min ago.",
      "saved_mins_ago": "Saved %{time} min ago.",
      "saved_hour_ago": "Saved %{time} hour ago.",
      "saved_hours_ago": "Saved %{time} hours ago.",
      "saved_day_ago": "Saved %{time} day ago.",
      "saved_days_ago": "Saved %{time} days ago.",
      "saving": "Saving...",
      "saving_impossible": "Impossible to save.",
      "new_attempt": "New attempt..."
    }
  },
  "Error": {
    "unshared_title": "This share is no longer active",
    "unshared_text": "The owner may have revoked this sharing. You can no longer edit a note at this address.",
    "loading_error_title": "The note couldn't be loaded",
    "loading_error_text_noReturnUrl": "The note may not exists anymore, or our servers may have a temporary glitch. Please try again later.",
    "loading_error_text_returnUrl": "The note may not exists anymore, or our servers may have a temporary glitch. Please go <a href='%{url}'>back</a> and try again later."
  },
  "manifest": {
    "short_description": "Cozy Notes is your personal and collaborative note-taking application.",
    "long_description": "Notes is an online text editor that can already offer you to:\n\n* Create your own notes with the essential features (bold, underlined, links, tables...)\n* Fill and find your notes with you other documents in files of your personal Drive\n* Edit easily your notes from all your devices\n* Access your notes offline\n* For the fans, use shortcuts and input in markdown format ... and even write with contacts together on the same note (coming soon)!",
    "screenshots": [
      "screenshots/en/screenshot01.png",
      "screenshots/en/screenshot02.png"
    ]
  }
}
