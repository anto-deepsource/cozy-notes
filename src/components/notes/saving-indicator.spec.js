import React from 'react'
import { I18n } from 'cozy-ui/react/I18n'
import { mount } from 'enzyme'
import SavingIndicator from './saving-indicator'
import en from '../../locales/en.json'

const sec = 1000
const min = 60 * sec
const hour = 60 * min
const day = 24 * hour

const getDirtySince = jest.fn()
const isDirty = () => !!getDirtySince()
const getLastSaveOrSync = jest.fn()
const on = jest.fn()

const collabProvider = {
  isDirty,
  getDirtySince,
  getLastSaveOrSync,
  on
}

function itMatchSnapshot(collabProvider) {
  it('should match snapshot', () => {
    const component = mount(
      <SavingIndicator collabProvider={collabProvider} />,
      {
        wrappingComponent: I18n,
        wrappingComponentProps: {
          lang: 'en',
          dictRequire: () => en
        }
      }
    )
    expect(component.html()).toMatchSnapshot()
  })
}

describe('SavingIndicator', () => {
  afterEach(() => jest.resetAllMocks())

  describe('when dirtySince is truthy', () => {
    describe('when dirtySince is a few minutes ago', () => {
      beforeEach(() => {
        getDirtySince.mockImplementation(() => new Date(Date.now() - 15 * min))
      })
      itMatchSnapshot(collabProvider)
    })
    describe('when dirtySince is a few seconds ago', () => {
      beforeEach(() => {
        getDirtySince.mockImplementation(() => new Date(Date.now() - 2 * sec))
      })
      itMatchSnapshot(collabProvider)
    })
  })

  describe('when dirtySince is falsy', () => {
    const steps = {
      'few seconds': 2 * sec,
      '45 sec': 45 * sec,
      '1 min': 1 * min,
      '5 min': 5 * min,
      '15 min': 15 * min,
      '28 min': 28 * min,
      '50 min': 50 * min,
      '1 hour': 1 * hour,
      '3 hour': 3 * hour,
      '1 day': 1 * day,
      '10 day': 10 * day
    }
    Object.keys(steps).forEach(key => {
      describe(`when lastSave is ${key} ago`, () => {
        beforeEach(() => {
          getLastSaveOrSync.mockImplementation(
            () => new Date(Date.now() - steps[key])
          )
        })
        itMatchSnapshot(collabProvider)
      })
    })
  })
})
