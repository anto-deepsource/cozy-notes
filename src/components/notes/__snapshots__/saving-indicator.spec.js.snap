// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SavingIndicator when dirtySince is falsy when lastSave is 1 day ago should match snapshot 1`] = `"<div class=\\"u-text saving-indicator\\">Up to date (saved 1 day ago)</div>"`;

exports[`SavingIndicator when dirtySince is falsy when lastSave is 1 hour ago should match snapshot 1`] = `"<div class=\\"u-text saving-indicator\\">Up to date (saved 1 hour ago)</div>"`;

exports[`SavingIndicator when dirtySince is falsy when lastSave is 1 min ago should match snapshot 1`] = `"<div class=\\"u-text saving-indicator\\">Up to date (saved 1 min. ago)</div>"`;

exports[`SavingIndicator when dirtySince is falsy when lastSave is 3 hour ago should match snapshot 1`] = `"<div class=\\"u-text saving-indicator\\">Up to date (saved 3 hours ago)</div>"`;

exports[`SavingIndicator when dirtySince is falsy when lastSave is 5 min ago should match snapshot 1`] = `"<div class=\\"u-text saving-indicator\\">Up to date (saved 5 min. ago)</div>"`;

exports[`SavingIndicator when dirtySince is falsy when lastSave is 10 day ago should match snapshot 1`] = `"<div class=\\"u-text saving-indicator\\">Up to date (saved 10 days ago)</div>"`;

exports[`SavingIndicator when dirtySince is falsy when lastSave is 15 min ago should match snapshot 1`] = `"<div class=\\"u-text saving-indicator\\">Up to date (saved 15 min. ago)</div>"`;

exports[`SavingIndicator when dirtySince is falsy when lastSave is 28 min ago should match snapshot 1`] = `"<div class=\\"u-text saving-indicator\\">Up to date (saved 20 min. ago)</div>"`;

exports[`SavingIndicator when dirtySince is falsy when lastSave is 45 sec ago should match snapshot 1`] = `"<div class=\\"u-text saving-indicator\\">Up to date (saved 40 sec. ago)</div>"`;

exports[`SavingIndicator when dirtySince is falsy when lastSave is 50 min ago should match snapshot 1`] = `"<div class=\\"u-text saving-indicator\\">Up to date (saved 50 min. ago)</div>"`;

exports[`SavingIndicator when dirtySince is falsy when lastSave is few seconds ago should match snapshot 1`] = `"<div class=\\"u-text saving-indicator\\">Up to date</div>"`;

exports[`SavingIndicator when dirtySince is truthy when dirtySince is a few minutes ago should match snapshot 1`] = `"<div class=\\"u-text saving-indicator\\">Changes unsaved</div>"`;

exports[`SavingIndicator when dirtySince is truthy when dirtySince is a few seconds ago should match snapshot 1`] = `"<div class=\\"u-text saving-indicator\\">Saving...</div>"`;
